File: AspNetMvcUnleashed/0672329980_source/Chapter14/VB/MvcApplication1/MvcApplication1/Controllers/MasterDetailController.vb
Public Class MasterDetailController
    Inherits Controller

    Private _entities As New ProductsDBEntities()

    Public Function Index() As ActionResult
        Return View(_entities.CategorySet.ToList())
    End Function


    Public Function Details(ByVal id As Integer) As ActionResult
        Dim products = From p In _entities.ProductSet _
                       Where p.CategoryId = id _
                       Select p
        Return PartialView("Details", products)
    End Function


End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter14/CS/MvcApplication1/MvcApplication1/Controllers/MasterDetailController.cs
using System.Collections.Generic;
using System.Linq;
using System.Web.Mvc;
using MvcApplication1.Models;

namespace MvcApplication1.Controllers
{

    public class MasterDetailController : Controller
    {
        private ProductsDBEntities _entities = new ProductsDBEntities(); 

        public ActionResult Index()
        {
            return View(_entities.CategorySet.ToList());
        }


        public ActionResult Details(int id)
        {
            var products = from p in _entities.ProductSet
                           where p.CategoryId == id
                           select p;
            return PartialView("Details", products);
        }


    }
}
